### PR TITLE
Spelling csharp

### PR DESCRIPTION
**Pull request by jsoref**
follow-up to https://github.com/github/codeql/pull/10743

### CODE REVIEW

File: csharp/ql/campaigns/Solorigate/lib/Solorigate.qll [hunk 1]
```diff
@@ -124,7 +124,7 @@ string solorigateSuspiciousLiterals() {
       "fc00::", "fe00::", "fec0::", "ffc0::", "ff00::", "HKCC", "HKCR", "HKCU", "HKDD",
       "HKEY_CLASSES_ROOT", "HKEY_CURRENT_CONFIG", "HKEY_CURRENT_USER", "HKEY_DYN_DATA",
       "HKEY_LOCAL_MACHINE", "HKEY_LOCAL_MACHINE\\SOFTWARE\\Microsoft\\Cryptography",
-      "HKEY_PERFOMANCE_DATA", "HKEY_USERS", "HKLM", "HKPD", "HKU", "If-None-Match",
+      "HKEY_PERFORMANCE_DATA", "HKEY_USERS", "HKLM", "HKPD", "HKU", "If-None-Match",
```
Comment (jsoref) [inline csharp/ql/campaigns/Solorigate/lib/Solorigate.qll:127]: This is notable

File: csharp/ql/src/experimental/Security Features/CWE-327/Azure/UnsafeUsageOfClientSideEncryptionVersion.cs [hunk 2]
```diff
@@ -17,7 +17,7 @@
     {
         KeyEncryptionKey = myKey,
         KeyResolver = myKeyResolver,
-        KeyWrapAlgorihm = myKeyWrapAlgorithm
+        KeyWrapAlgorithm = myKeyWrapAlgorithm
```
Comment (jsoref) [inline csharp/ql/src/experimental/Security Features/CWE-327/Azure/UnsafeUsageOfClientSideEncryptionVersion.cs:20]: This seems problematic

File: csharp/old-change-notes/2020-08-26-implicit-array-lengths.md [hunk 3]
```diff
@@ -1,5 +1,5 @@
 lgtm,codescanning
-* Inferring the lengths of implicitely sized arrays is fixed. Previously, multi
+* Inferring the lengths of implicitly sized arrays is fixed. Previously, multi
```
Comment (michaelnebel) [inline csharp/old-change-notes/2020-08-26-implicit-array-lengths.md:2]: We probably shouldn't change anything in the old change notes.

File: csharp/downgrades/initial/semmlecode.csharp.dbscheme [hunk 4]
```diff
@@ -245,7 +245,7 @@ tokens(
               | @destructor | @callable_accessor | @value_or_ref_type | @declaration_with_accessors
               | @local_function;
 
-/** LOCATIONS, ASEMMBLIES, MODULES, FILES and FOLDERS **/
```
Comment (michaelnebel) [inline csharp/downgrades/initial/semmlecode.csharp.dbscheme:248]: Even though there are spelling errors we shouldn't change anything in the `dbscheme` files as the hash of these files are used to determine upgrade/downgrade order.

File: csharp/ql/lib/experimental/code/csharp/Cryptography/NonCryptographicHashes.qll [hunk 5]
```diff
@@ -10,7 +10,7 @@ import csharp
 private import DataFlow
 private import semmle.code.csharp.dataflow.TaintTracking2
 
-predicate maybeANonCryptogrphicHash(Callable callable, Variable v, Expr xor, Expr mul, LoopStmt loop) {
+predicate maybeANonCryptographicHash(Callable callable, Variable v, Expr xor, Expr mul, LoopStmt loop) {
```
Comment (michaelnebel) [inline csharp/ql/lib/experimental/code/csharp/Cryptography/NonCryptographicHashes.qll:13]: This is a publicly exposed predicate, so the predicate with the incorrect spelling needs to be deprecated and not removed directly.

File: csharp/ql/src/experimental/Security Features/CWE-327/Azure/UnsafeUsageOfClientSideEncryptionVersion.cs (hunk 2 above)
Comment (michaelnebel) [reply csharp/ql/src/experimental/Security Features/CWE-327/Azure/UnsafeUsageOfClientSideEncryptionVersion.cs:20]: +1

**Review by michaelnebel — CHANGES_REQUESTED** [review]
Really nice to get some of the spelling corrected.
There are a couple of places, where we unfortunately can't correct the spelling.

File: csharp/downgrades/193c35b0e3776f8ee9a49e984e393974b4c670f6/old.dbscheme [hunk 6]
```diff
@@ -245,7 +245,7 @@ tokens(
               | @destructor | @callable_accessor | @value_or_ref_type | @declaration_with_accessors
               | @local_function;
 
-/** LOCATIONS, ASEMMBLIES, MODULES, FILES and FOLDERS **/
+/** LOCATIONS, ASSEMBLIES, MODULES, FILES and FOLDERS **/
```
Comment (hvitved) [inline csharp/downgrades/193c35b0e3776f8ee9a49e984e393974b4c670f6/old.dbscheme:248]: Please revert all changes to files in `csharp/downgrades` and `csharp/upgrades`.

File: csharp/ql/lib/experimental/code/csharp/Cryptography/NonCryptographicHashes.qll (hunk 5 above)
Comment (hvitved) [reply csharp/ql/lib/experimental/code/csharp/Cryptography/NonCryptographicHashes.qll:13]: Not needed, I think, as this is in `experimental`.

File: csharp/ql/lib/semmlecode.csharp.dbscheme [hunk 7]
```diff
@@ -199,7 +199,7 @@ sourceLocationPrefix(
               | @destructor | @callable_accessor | @value_or_ref_type | @declaration_with_accessors
               | @local_function | @lambda_expr;
 
-/** LOCATIONS, ASEMMBLIES, MODULES, FILES and FOLDERS **/
+/** LOCATIONS, ASSEMBLIES, MODULES, FILES and FOLDERS **/
```
Comment (hvitved) [inline csharp/ql/lib/semmlecode.csharp.dbscheme:202]: I suggest to revert this change, as we would otherwise need to create upgrade/downgrade scripts.

**Review by hvitved — CHANGES_REQUESTED** [review]
Thanks!

File: csharp/old-change-notes/2020-08-26-implicit-array-lengths.md (hunk 3 above)
Comment (jketema) [reply csharp/old-change-notes/2020-08-26-implicit-array-lengths.md:2]: We have been accepting spelling fixes to change notes in some of the other PRs that were opened, for example, [here](https://github.com/github/codeql/pull/10802/files#diff-784d6a6c025528388cf2190acef01f5076a9ef9e597baf78d9e9d911016c7fb2) and [here](https://github.com/github/codeql/pull/10801/files#diff-218bb605d541548f68e8ef510ee7de4c94215c559d577071fd9fc753c1a25a88), although those changes were not in `old-change-notes` directories. The change note will now of course be slightly out-of-sync with the one in the released version, but not in a way that seems harmful. Unless I completely overlooked something?

**Review by hvitved — APPROVED** [review]

File: csharp/ql/src/utils/model-generator/internal/CaptureModels.qll [hunk 8]
```diff
@@ -247,7 +247,7 @@ string captureSource(DataFlowTargetApi api) {
  * A TaintTracking Configuration used for tracking flow through APIs.
  * The sources are the parameters of the API and the fields of the enclosing type.
  *
- * This can be used to generate Sink summaries for APIs, if the API propgates a parameter (or enclosing type field)
+ * This can be used to generate Sink summaries for APIs, if the API propagates a parameter (or enclosing type field)
```
Comment (hvitved) [inline csharp/ql/src/utils/model-generator/internal/CaptureModels.qll:250]: This file and `java/ql/src/utils/model-generator/internal/CaptureModels.qll` need to be in sync, so you need to update them on the same PR (e.g. this PR).

File: csharp/ql/lib/experimental/code/csharp/Cryptography/NonCryptographicHashes.qll [hunk 9]
```diff
@@ -10,7 +10,7 @@ import csharp
 private import DataFlow
 private import semmle.code.csharp.dataflow.TaintTracking2
```
Comment (hvitved) [inline csharp/ql/lib/experimental/code/csharp/Cryptography/NonCryptographicHashes.qll:11]: This file is no longer properly auto-formatted.

**Review by hvitved — CHANGES_REQUESTED** [review]

**Review by hvitved — APPROVED** [review]
